R/toml_serialization.R

```diff
@@ -114,9 +113,10 @@ simplify_row <- function(row) {
 format_toml <- function(x, ..., .top_level = FALSE) UseMethod("format_toml")
 
 format_toml.default <- function(x, ..., .top_level = FALSE) {
-  ui_throw(
+  cli::cli_abort(c(
     get_toml_err_msg(),
-    bullet_x("`{typeof(x)}` cannot be converted to toml.")
+    "x" = "`{class(x)}` cannot be converted to toml."
```

Comment:
This change seems correct, since we're using S3 to dispatch to TOML? @Ilia-Kosenkov can you confirm?

Reply:
Yep, we rely on `S3`. Is it worth outputting whole `class()` (could be a vector)? Also, I suspect it should be some `{.code {class(x)}}`.

Reply:
Yeah. This was a goofy one. Take the example here:

```r
to_toml(cat = structure("Meow", class = "kitty"))
Error in `map2_chr()` at rextendr/R/toml_serialization.R:55:2:
ℹ In index: 1.
Caused by error in `format_toml()` at rextendr/R/toml_serialization.R:57:4:
! Object cannot be serialized.
✖ `kitty` cannot be converted to toml.
```

Previously this would said character cannot be converted to toml. which would be wrong :O

Reply:
Resolved

``` r
rextendr::to_toml(cat = structure("Meow", class = c("kitty", "cat")))
#> Error in `map2_chr()` at rextendr/R/toml_serialization.R:55:2:
#> ℹ In index: 1.
#> Caused by error in `format_toml()` at rextendr/R/toml_serialization.R:57:4:
#> ! Object cannot be serialized.
#> ✖ `kitty` and `cat` cannot be converted to toml.

#> Backtrace:
#>      ▆
#>   1. └─rextendr::to_toml(...)
#>   2.   └─purrr::map2_chr(...) at rextendr/R/toml_serialization.R:55:2
#>   3.     └─purrr:::map2_("character", .x, .y, .f, ..., .progress = .progress)
#>   4.       ├─purrr:::with_indexed_errors(...)
#>   5.       │ └─base::withCallingHandlers(...)
#>   6.       ├─purrr:::call_with_cleanup(...)
#>   7.       └─rextendr (local) .f(.x[[i]], .y[[i]], ...)
#>   8.         ├─rextendr:::format_toml(...) at rextendr/R/toml_serialization.R:57:4
#>   9.         └─rextendr:::format_toml.default(...) at rextendr/R/toml_serialization.R:113:15
#>  10.           └─cli::cli_abort(c(get_toml_err_msg(), x = "{.code {class(x)}} cannot be converted to toml.")) at rextendr/R/toml_serialization.R:116:2
#>  11.             └─rlang::abort(...)
```

<sup>Created on 2023-04-17 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>